mesheryctl/pkg/utils/sse-client_test.go

```diff
@@ -0,0 +1,140 @@
+package utils
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+func TestConvertRespToSSE(t *testing.T) {
+	tests := []struct {
+		name string
+		body string
+	}{
+		{
+			name: "Convert Response to SSE with sample body",
+			body: "Test data",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			resp := http.Response{
+				Body: ioutil.NopCloser(bytes.NewBufferString(tt.body)),
+			}
+
+			_, err := ConvertRespToSSE(&resp)
+
+			if err != nil {
+				t.Fatalf("ConvertRespToSSE() error = %v", err)
+				return
+			}
+		})
+	}
+}
+
+func Test_loop(t *testing.T) {
+	tests := []struct {
+		name    string
+		body    string
+		fixture string
+		golden  string
+	}{
+		{
+			name:    "Successful SMI test event",
+			body:    "data: {\"summary\":\"Smi conformance test completed successfully\",\"details\":\"Result-Id: ce336fe4-6e01-4779-949b-6526e44e3d55\",\"operation_id\":\"b33e2b2d-4826-4ea7-8c41-f8f76e8e4bf7\"}\nid: test1\nevent: testevent1\n:\"Comment to be ignored\"\n\n",
```

Comment:
why test loop separately when this function is already being called through `ConvertRespToSSE`

Reply:
Hey, sorry for the late response. So for keeping with the spirit of unit testing, I thought that in the unit test for ```ConvertRespToSSE``` I'd just check the immediate return value of the function, and test the ```loop``` goroutine being called there separately. So in this unit test I'm testing the actual behaviour of ```loop``` . This way we're testing both the functions, and the coverage increases too.

Let me know if this is okay or if any changes are required :+1: